src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalTypeCheck.java

```diff
@@ -289,6 +289,39 @@
  * }
  * </pre>
  * <p>
+ * To configure the check to target fields types only:
+ * </p>
+ * <pre>
+ * &lt;module name="IllegalType"&gt;
+ *   &lt;property name=&quot;illegalClassNames&quot; value=&quot;java.util.Optional&quot;/&gt;
+ *   &lt;property name=&quot;tokens&quot; value=&quot;VARIABLE_DEF&quot;/&gt;
+ *   &lt;property name=&quot;id&quot; value=&quot;IllegalTypeOptionalAsField&quot;/&gt;
+ * &lt;/module&gt;
+ * &lt;module name="SuppressionXpathSingleFilter"&gt;
+ *   &lt;property name=&quot;query&quot; value=&quot;//METHOD_DEF//*&quot;/&gt;
+ *   &lt;property name=&quot;id&quot; value=&quot;IllegalTypeOptionalAsField&quot;/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <pre>
+ * import java.util.Optional;
+ *
+ * public class Main {
+ *
+ *   private Optional&lt;String&gt; field1; // violation, usage of type 'Optional' is not allowed
+ *   static int field2 = 4; // OK
+ *   protected String field3; // OK
+ *   public Optional&lt;String&gt; field4; // violation, usage of type 'Optional' is not allowed
+ *   Optional&lt;String&gt; field5; // violation, usage of type 'Optional' is not allowed
+ *
+ *   void foo() {
+ *     Optional&lt;String&gt; i; // OK
+ *   }
+ *   public &lt;T extends java.util.Optional&gt; void Method(T t){
```

Comment:
1. `Method` -> `method`
2. `t){` <- whitespace before `{`
3. Please show that bounded type parameter is OK too, since IllegalType also checks this.

Reply:
> Please show that bounded type `parameter` is OK too, since IllegalType also checks this.

but we are only targeting fields types here?

Reply:
> but we are only targeting fields types here?

Then why show the local variables as `// OK`?  This method does not add to the example if we do not show why this method and type parameter exist in this case.

Reply:
yea i got that that's why i deleted that comment :)
is that what u meant
```
public <T extends java.util.Optional> void method(T t) {
    Optional<T> i = t; // OK
  }
```

Reply:
IllegalType also checks type parameters in method signatures, specifically the one on the line that this comment is on.  You should place an `// OK` comment on this line.  Take a look at the other examples in the documentation for this check with type parameters in method signatures.

Reply:
when i use `<property name="tokens" value="PARAMETER_DEF"/>` i get an error in 1) 
```
1)  public <T extends java.util.Optional> void method(T t, Optional<T> j) { // OK
2)    Optional<T> i; // OK
  }
```

Reply:
done

Reply:
Resolved.